### ISSUE TITLE
Package installation fails when ZIP file has a specified version

### ISSUE THREAD

**Issue by danielbachhuber** [wp-cli/wp-cli#3518]
```
$ package install profile.0.3.0-beta.zip
Installing package runcommand/profile (dev-master)
Updating /Users/danielbachhuber/salty-wordpress/wp-cli/packages/composer.json to require the package...
Registering /Users/danielbachhuber/salty-wordpress/wp-cli/packages/local/runcommand-profile as a path repository...
Using Composer to install the package...
---
Loading composer repositories with package information
Updating dependencies
Resolving dependencies through SAT
Dependency resolution completed in 0.283 seconds
Your requirements could not be resolved to an installable set of packages.
Problem 1
    - The requested package runcommand/profile dev-master exists as runcommand/profile[0.3.0-beta] but these are rejected by your constraint.
---
Error: Package installation failed (Composer return code 2). Reverted composer.json
```

**Comment by schlessera** [comment]
What is your `"minimum-stability"` set to? The version that would have been pulled in is marked as a pre-release version, so the `"minimum-stability"` needs to be set to a level that supports that: https://getcomposer.org/doc/04-schema.md#minimum-stability
If you do that, remember to set the `"prefer-stable"` to true as well, to avoid pulling in development versions for everything...

**Comment by danielbachhuber** [comment]
> What is your "minimum-stability" set to?

It's set to `dev` in `~/.wp-cli/packages/composer.json`...

**Comment by schlessera** [comment]
Composer seems to use `/Users/danielbachhuber/salty-wordpress/wp-cli/packages/composer.json` to pull in this package...

**Comment by danielbachhuber** [comment]
Sorry, those are one and the same.

**Comment by schlessera** [comment]
Can you post the final `composer.json` that is being used? I think it is modified on the fly, right?

**Comment by danielbachhuber** [comment]
> I think it is modified on the fly, right?

This was just the hint I needed :) Fixed the error of my ways in #3519. Thanks!
